### PR TITLE
PS-6095 [Doc] Port 8.0 Security folder to 5.7

### PR DESCRIPTION
**Pull request by patrickbirch**
Added security folder
removed data-masking
updating using-keyring
rewrote and updated master key overview and enabling master key (which provides samples).
Added **experimental** when needed.
Added address to Management/data-at-rest encryption

### CODE REVIEW

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/enabling-master-key.rst]: innodb_default_encryption_key_id is part of KEYRING encryption, should not be here.

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/enabling-master-key.rst]: This is true when innodb_file_per_table is set to OFF. If file per table is on, you do not need system tablespace to be encrypted when creating new tables.

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/enabling-master-key.rst:105]: This should be marked as experimental - wasn't it marked experimental previously ?

File: doc/source/security/overview-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/overview-master-key.rst]: This is not part of Master Key encryption, but Keyring encryption.

File: doc/source/security/overview-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/overview-master-key.rst]: this feature is GA for ~4 years now.

**Review by robgolebiowski — CHANGES_REQUESTED** [review]

File: doc/source/security/enabling-master-key.rst
Comment (patrickbirch) [reply doc/source/security/enabling-master-key.rst]: removed.

File: doc/source/security/enabling-master-key.rst
Comment (patrickbirch) [reply doc/source/security/enabling-master-key.rst]: Updated. Changed to OFF. Removed ``bootstrap`` references and added ``initialize``.

File: doc/source/security/enabling-master-key.rst
Comment (patrickbirch) [reply doc/source/security/enabling-master-key.rst:105]: added experimental.

File: doc/source/security/overview-master-key.rst
Comment (patrickbirch) [reply doc/source/security/overview-master-key.rst]: section removed.

File: doc/source/security/overview-master-key.rst
Comment (patrickbirch) [reply doc/source/security/overview-master-key.rst]: removed experimental. Time flies.

**Comment by patrickbirch** [comment]
waiting for comments or approval

File: doc/source/security/enabling-master-key.rst
Comment (george-lorch) [reply doc/source/security/enabling-master-key.rst:105]: This is/was supposed to be GA in 5.7 as it is now 'inherited' from MySQL 8.0 and backported.

Comment (robgolebiowski) [reply doc/source/security/enabling-master-key.rst:105]: @georgelorchpercona rotate_system_key is part of our encryption keys rotation, so it was not inherited from upstream.

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/enabling-master-key.rst:30]: I am not sure why is it here ? Is it a link ? Because here we talk about encrypting a general tablespace and not about file-per-table.

Comment (robgolebiowski) [reply doc/source/security/enabling-master-key.rst:30]: Or you are describing file_per_table - then file_per_table is not a general tablespace.

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/enabling-master-key.rst:92]: binary log encryption is not an experimental feature, it is GA

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [reply doc/source/security/enabling-master-key.rst:105]: @patrickbirch @georgelorch I see - you added binlog encryption as experimental - I meant just You can rotate the encryption (...) SELECT rotate_system_key("percona_binlog") <- this is still experimental.

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/enabling-master-key.rst:149]: @patrickbirch @georgelorchpercona as it was decided that we are going to remove the encryption threads from 5.7 and INFORMATION_SCHEMA.INNODB_TABLESPACE_ENCRYPTION is part of the encryption threads WL - shouldn't the information on INNODB_TABLESPACE_ENCRYPTION be removed ? I think yes.

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/enabling-master-key.rst:98]: in 5.7 binlog encryption is enabled by enabling –encrypt-binlog=1. –master_verify_checksum and –binlog_checksum both have to be ON (see https://www.percona.com/blog/2018/03/08/binlog-encryption-percona-server-mysql/ for more details)

File: doc/source/security/enabling-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/enabling-master-key.rst:186]: this 8.0 only feature - the Encrypted column in SHOW BINARY LOGS output.

File: doc/source/security/overview-master-key.rst
Comment (robgolebiowski) [inline doc/source/security/overview-master-key.rst:281]: As I have commented earlier this is enabled by setting encrypt_binlog. Later in this paragraph, you describe correctly encrypt_binlog variable.

**Review by robgolebiowski — CHANGES_REQUESTED** [review]

File: doc/source/security/enabling-master-key.rst
Comment (george-lorch) [reply doc/source/security/enabling-master-key.rst:105]: yes, I was speaking specifically to binlog encryption, not rotation

File: doc/source/security/enabling-master-key.rst
Comment (george-lorch) [reply doc/source/security/enabling-master-key.rst:149]: agreed

File: doc/source/security/enabling-master-key.rst
Comment (patrickbirch) [reply doc/source/security/enabling-master-key.rst:105]: Moved the Experimental tag to the rotate_system_key for the binlog.

File: doc/source/security/enabling-master-key.rst
Comment (patrickbirch) [reply doc/source/security/enabling-master-key.rst:30]: I was attempting to link from the overview-master-key doc. Removed. Will update overview-master-key doc.

File: doc/source/security/enabling-master-key.rst
Comment (patrickbirch) [reply doc/source/security/enabling-master-key.rst:92]: Moved to the description of the key rotation.

File: doc/source/security/enabling-master-key.rst
Comment (patrickbirch) [reply doc/source/security/enabling-master-key.rst:149]: Removed the section about the I_S.Innodb_tablespace_encryption description.

File: doc/source/security/enabling-master-key.rst
Comment (patrickbirch) [reply doc/source/security/enabling-master-key.rst:186]: Removed information and the example for the SHOW BINARY LOGS.

**Comment by patrickbirch** [comment]
Will be replaced

**Pull request closed without merging**
